pennylane/resource/error.py

```diff
@@ -77,3 +78,72 @@ def error(self) -> AlgorithmicError:
         Returns:
             AlgorithmicError: The error.
         """
+
+
+class SpectralNormError(AlgorithmicError):
+    """Class representing the spectral norm error.
+
+    The spectral norm error is defined as the distance, in spectral norm, between the true unitary
+    we intend to apply and the approximate unitary that is actually applied.
+
+    Args:
+        error (float): The numerical value of the error
+
+    **Example**
+
+    >>> s1 = SpectralNormError(0.01)
+    >>> s2 = SpectralNormError(0.02)
+    >>> s3 = s1.combine(s2)
+    >>> print(s3, s3.error)
+    <SpectralNormError(0.03)> 0.03
+    """
+
+    def __repr__(self):
+        """Return formal string representation."""
+
+        return f"<SpectralNormError({self.error})>"
```

Comment:
@DSGuala any preference on how this looks?
I believe this would be better, as it allows the output to be copied as valid code.
```suggestion
        return f"SpectralNormError({self.error})"
```

Reply:
Yup, I prefer this new version too :+1:

Reply:
Updated.